### PR TITLE
Transformation sting wears off after time

### PR DESCRIPTION
**Pull request by ItsTisme**
## About The Pull Request

Transformation sting will now dissapear after X amount of time(6000 currently)
## Why It's Good For The Game

ided

Its pretty annoying to get transform stung into glass golem/felind/IPC round start with no realistic way to revert it and no real way to defend against it in the first place. Atleast being able to turn back makes it not used only to cheese and meme

## Changelog
:cl:
balance: Transformation sting will undo itself after certain amount of time
/:cl:

### CODE REVIEW

**Comment by bloons3** [comment]
This entirely guts the ability though

**Comment by ItsTisme** [comment]
> This entirely guts the ability though

Does it though? We can extend the time to 5-10 minuts but I dont even see it used for decieving currently. Its only used to do le funny and transform as many people as possible into yourself

File: code/modules/antagonists/changeling/powers/tiny_prick.dm
Comment (Archanial) [inline code/modules/antagonists/changeling/powers/tiny_prick.dm]:
```suggestion
	var/datum/dna/new_dna = selected_dna.dna
```

File: code/modules/antagonists/changeling/powers/tiny_prick.dm
Comment (Archanial) [inline code/modules/antagonists/changeling/powers/tiny_prick.dm]:
```suggestion
		var/datum/dna/old_dna = new /datum/dna
```

Comment (Archanial) [reply code/modules/antagonists/changeling/powers/tiny_prick.dm]: On second thought do you really need to initialize this DNA if you are just doing to copy it in next line?

File: code/modules/antagonists/changeling/powers/tiny_prick.dm
Comment (Archanial) [inline code/modules/antagonists/changeling/powers/tiny_prick.dm]:
```suggestion
/datum/action/changeling/sting/transformation/proc/remove_transform(mob/target, var/datum/dna/old_self)
```

**Comment by Archanial** [comment]
Bad idea. It kills the ability entirely. Being turned into someone else is really rare and should remain "hard to cure".

**Comment by OceanFish1** [comment]
Then remove, don't improve the ability :trollface:

**Comment by Sinestia** [comment]
> Atleast being able to turn back makes it not used only to cheese and meme

it probably won't be used at all then


i think it's fine as is, usually it's a benefit to rounds aside from a few salty carebears

**Comment by ghost** [comment]
Makes the ability useless

**Comment by super12pl** [comment]
You can still make people drop their armors and be unable to use weapons with it
As well as fucking 2 hit dust them

**Comment by PowerfulBacon** [comment]
2 minutes of turning someone else into someone else seems completely useless. You don't even need to be a changeling to temporarily transform into someone else, just go to genetics.

File: code/modules/antagonists/changeling/powers/tiny_prick.dm
Comment (ItsTisme) [reply code/modules/antagonists/changeling/powers/tiny_prick.dm]: Yes, for a deep copy. Since old_dna = target, when I change target's DNA it will stay the same since the target dna = new dna but also old dna because its just reference

**Comment by ItsTisme** [comment]
> 2 minutes of turning someone else into someone else seems completely useless. You don't even need to be a changeling to temporarily transform into someone else, just go to genetics.

I could make it longer, 2 minuts was just something random iam willing to make it 10 or more

**Comment by ItsTisme** [comment]
> You can still make people drop their armors and be unable to use weapons with it
> As well as fucking 2 hit dust them

I might make a dif PR to make it so you you cant get golem'ed. Gotta atomize

**Comment by ItsTisme** [comment]
> Then remove, don't improve the ability :trollface:

I had the idea to instead make it Transform sting in a AoE and make smoke  cloud to hide amongst the crowd. Making it way more viable as deceiving tactic

**Comment by ItsTisme** [comment]
its now 6000

**Comment by WhyIsCaeciliusTaken** [comment]
Don't do this. Give TRAIT_NOTRANSSTING to glass golems etc instead

**Comment by eeSPee** [comment]
> Don't do this. Give TRAIT_NOTRANSSTING to glass golems etc instead

But that would mean improve, dont hardnerf.

Goes against the philosophy of bee.

**Comment by ItsTisme** [comment]
> Don't do this. Give TRAIT_NOTRANSSTING to glass golems etc instead

I mean you also have sand golems with roughly same problem, or cloth golems etc. And in general becoming golem makes you drop your armor and prevents you from using any guns. Among other epic funny species abuses

**Comment by eeSPee** [comment]
> > Don't do this. Give TRAIT_NOTRANSSTING to glass golems etc instead
> 
> I mean you also have sand golems with roughly same problem, or cloth golems etc. And in general becoming golem makes you drop your armor and prevents you from using any guns. Among other epic funny species abuses

So exclude golems, the same way IPCs and plasmemes are excluded.

**Comment by PowerfulBacon** [comment]
> 
> 
> > Don't do this. Give TRAIT_NOTRANSSTING to glass golems etc instead
> 
> I mean you also have sand golems with roughly same problem, or cloth golems etc. And in general becoming golem makes you drop your armor and prevents you from using any guns. Among other epic funny species abuses

Then add NOTRANSSTING to golems.

**Review by utilbird — CHANGES_REQUESTED** [review]
above

**Comment by wesoda25** [comment]
> > This entirely guts the ability though
> 
> 
> 
> Does it though? We can extend the time to 5-10 minuts but I dont even see it used for decieving currently. Its only used to do le funny and transform as many people as possible into yourself

You're going about the wrong way of fixing it then. This change nerfs both the intended and unintended use of the ability. 

If grief is your complaint, then address it. You could add a limit to how many people can be transformed to prevent mass conversion, or possibly a short do after so that it's less easy to sting the whole station. 

I don't see how just overall making the ability worse is a good solution.

**Comment by ItsTisme** [comment]
> above

figurd that was more fitting for dif PR entirly but ok?

**Comment by ItsTisme** [comment]
>You could add a limit to how many people can be transformed to prevent mass conversion

Keeping track of people and there DNA for this is both some what pricy to do and also kinda janky. If people randomly transform back it could confirm that a very specific ling is still alive which makes transforming to fake your own death pointless.

Iam open for suggestions on how to better do this I just want A way to revert it(without metaing gentics to save your DNA) and this was the easiest to make

**Comment by Sinestia** [comment]
Maybe mutadone reverting it could be workable

**Comment by bloons3** [comment]
Apply NOSTING to golems and IPCs, since they are not organic beings and do not have DNA.

**Comment by ItsTisme** [comment]
> Maybe mutadone reverting it could be workable

I thought that mutadone was too common/ easily makable for this purpose

**Comment by ItsTisme** [comment]
> Apply NOSTING to golems and IPCs, since they are not organic beings and do not have DNA.

Golems being unstingable is a follow PR I wana do cuz atomizing. Though dont think gameplay wise its fun IPC's get even more meta antag protection they they already due by virtue of being shit code

**Comment by Sinestia** [comment]
I don't think it's that bad. Especially if ruko hits it in his chem
rework


 Main issue with mutadone off the top of my head I think is that 0.1u
is adequate for just purging mutations unless I'm mistaken. You could
avoid that by requiring 10u or so.

**Comment by bloons3** [comment]
I mean, the OTHER method of changing your identity is to use the gene machine in genetics which ALSO permanently changes your appearance.

**Comment by ItsTisme** [comment]
> I mean, the OTHER method of changing your identity is to use the gene machine in genetics which ALSO permanently changes your appearance.

True but its unreasonble to get dragged to genetics and get this forced on you since you can fight back and is super obviuse.

Unlike lings who can bypass armor and do it on mass, which is what this PR wanted to adress

**Comment by Sinestia** [comment]
Spamming UEs on people is a lot more high profile and liable to get
your ass beat

**Comment by bloons3** [comment]
> > I mean, the OTHER method of changing your identity is to use the gene machine in genetics which ALSO permanently changes your appearance.
> 
> True but its unreasonble to get dragged to genetics and get this forced on you since you can fight back and is super obviuse.
> 
> Unlike lings who can bypass armor and do it on mass, which is what this PR wanted to adress

Then have it check armor, or other things that would reasonably stop stings.

**Comment by wesoda25** [comment]
>If people randomly transform back it could confirm that a very specific ling is still alive which makes transforming to fake your own death pointless.

.... This is an argument against your own PR. 

Also, it wouldn't be pricey at all. Just use a simple ++ or -- every time they sting someone, and cap it at a number you think is appropriate. You could even allow them to increase this limit by absorbing other lings.

**Comment by super12pl** [comment]
> Spamming UEs on people is a lot more high profile and liable to get
your ass beat

They still will manage to sting half the station before being caught and cremated.

**Comment by ItsTisme** [comment]
> Also, it wouldn't be pricey at all. Just use a simple ++ or -- every time they sting someone, and cap it at a number you think is appropriate. You could even allow them to increase this limit by absorbing other lings.

It would, because I have to make hard copy's of every orignal DNA. So either every carbon gets a new var to container there orignal DNA, or the ling has to keep track of 5-10 people's DNA to then revert them. Also 10 minut timer should be enough and will not opt a ling out because it will revert regardless if the ling is alive or not. Unlike a hardcap which will confirm a ling is alive because he stung some one else to cause you to revert

**Comment by utilbird** [comment]
I can't really see any good reasons to nerf transformation sting, comparing it to the (slightly larger) effort you need to do in genetics isn't fair when you factor in that transformation is an _antag ability_.

**Comment by super12pl** [comment]
> I can't really see any good reasons to nerf transformation sting, comparing it to the (slightly larger) effort you need to do in genetics isn't fair when you factor in that transformation is an _antag ability_.

> slightly larger

**Comment by utilbird** [comment]
Slightly was probably a bad word. But it's not _that_ hard

**Comment by utilbird** [comment]
Either way, transformation sting isn't really strong enough to warrant a nerf, this is one of the few purely deceptive abilities a ling has, a restoration timer would make it practically useless. obligatory: feel free to re-open if you can get another maintainer's approval for the PR.

**Comment by ghost** [comment]
clings are already useless smh

**Pull request closed without merging**
